armsrc/sam_picopass.c

```diff
@@ -30,6 +30,213 @@
 #include "protocols.h"
 #include "optimized_cipher.h"
 #include "fpgaloader.h"
+#include "pm3_cmd.h"
+
+/**
+ * @brief Sends a request to the SAM and retrieves the response.
+ *
+ * Unpacks request to the SAM and relays ISO15 traffic to the card.
+ * If no request data provided, sends a request to get PACS data.
+ *
+ * @param request Pointer to the buffer containing the request to be sent to the SAM.
+ * @param request_len Length of the request to be sent to the SAM.
+ * @param response Pointer to the buffer where the retreived data will be stored.
+ * @param response_len Pointer to the variable where the length of the retreived data will be stored.
+ * @return Status code indicating success or failure of the operation.
+ */
+static int sam_send_request_iso15(const uint8_t *const request, const uint8_t request_len, uint8_t *response, uint8_t *response_len, const bool shallow_mod, const bool break_on_nr_mac, const bool prevent_epurse_update) {
+    int res = PM3_SUCCESS;
+    if (g_dbglevel >= DBG_DEBUG)
+        DbpString("start sam_send_request_iso14a");
+
+    uint8_t buf1[ISO7816_MAX_FRAME] = {0};
+    uint8_t buf2[ISO7816_MAX_FRAME] = {0};
```

Comment:
consider making these "bigbuf_malloc"  and free the memory once the whole command is ran.

Reply:
Changed here and in SEOS. I think I carried this static allocation from some previous version of my code, where I tried using BigBuf for more temporary buffers.